### PR TITLE
PLT-2664 Split out push notifications from email notifications

### PR DESCRIPTION
**Pull request by jwilander**
Also completes PLT-1577.

### CODE REVIEW

File: api/post.go [hunk 1]
```diff
@@ -576,186 +569,64 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
...
-		for id, doSend := range toEmailMap {
+	for id := range mentionedUserIds {
+		mentionedUsersList = append(mentionedUsersList, id)
+	}
 
-			if !doSend {
-				continue
+	if utils.Cfg.EmailSettings.SendEmailNotifications {
+		for _, id := range mentionedUsersList {
+			userAllowsEmails := true
```
Comment (hmhealey) [inline api/post.go]: Couldn't this just be `userAllowsEmails := profileMap[id].NotifyProps["email"] != "false"`? If they key doesn't exist in NotifyProps, it'll just return the empty string

**Comment by hmhealey** [comment]
+1 other than that tiny style thing. That sendNotifications function looks a lot cleaner than it did before

**Comment by coreyhulen** [comment]
+1

File: api/post.go (hunk 1 above)
Comment (jwilander) [inline api/post.go]: Uuuuh yes, I think you're right. Updating..

**Pull request merged**